src/applications/vaos/hooks/useFormState.jsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * @module hooks
+ */
+import { useState, useEffect } from 'react';
+import { getDefaultFormState } from '@department-of-veterans-affairs/react-jsonschema-form/lib/utils';
+
+import {
+  updateSchemaAndData,
+  updateItemsSchema,
+} from 'platform/forms-system/src/js/state/helpers';
+
+function setupFormData(data, schema, uiSchema) {
+  const schemaWithItemsCorrected = updateItemsSchema(schema);
+  return updateSchemaAndData(
+    schemaWithItemsCorrected,
+    uiSchema,
+    getDefaultFormState(schemaWithItemsCorrected, data, {}),
+  );
+}
+
+/**
+ * Hook for managing the state for a SchemaForm page in local component state
+ * instead of redux
+ *
+ * @export
+ * @name useFormState
+ * @function
+ * @param {Object} params Hook parameters
+ * @param {Object|Function} params.initialSchema The initial schema for the form. Can also be a function, which will
+ *   only be executed on startup or when enabled or dependencies changes
+ * @param {Object} params.uiSchema The uiSchema for the form
+ * @param {Object} params.initialData The initial data to use for the form state. This is only used the first time the
+ *   hook is called
+ * @param {Array} [params.dependencies=[]] Array of useEffect dependencies that will force the form state to be reset using the current
+ *   values for initialSchema and uiSchema.
+ * @param {boolean} [params.enabled=true] Indicates if the form state is enabled or not. All values will be returned as null
+ *   when this is false, and when it changes to true the form state will be set up using the current schema and uiSchema
+ *   values
+ * }
+ * @returns {{ schema: Object, uiSchema: Object, data: Object }} The form state values, to be used with SchemaForm
+ */
+export default function useFormState({
```

Comment:
This hook might be hard to follow, but it reuses the form system functions we call in our Redux actions to set up and change form state. In addition it provides some mechanisms for making sure we don't run our initial schema update code over and over (`initialSchema` can be a function) and to allow us to update our form after data changes later (like after an api call) with `dependencies` and `enabled`.